.github/workflows/build-apps.yml

```diff
@@ -22,6 +22,8 @@ jobs:
 
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
+        with:
+          gradle-home-cache-cleanup: true
```

Comment:
I think this is the only one that actually runs on main.

Reply:
Makes sense. That also means if a PR workflow uses a big dependency, the next MAIN workflow will most likely delete it because it wasn't used during the run. I think that's acceptable as MAIN should do most of the work. Anyways, I'll remove the other ones.